dev-docs/RFCs/v4.0.0/polyfill-modes-rfc.md

```diff
@@ -12,10 +12,79 @@ Our current polyfill algorithm allocates cells to polygons based on whether the
 
 *Why is this important?*
 
+* Inclusion test
+
+A use case we want to support is testing whether a point could possibly be in a polygon by set containment. In Java this would look like:
+
+```java
+final int R = ...;
+// Indexed as res R
+Set<String> cellsCoveringPolygon = ...;
+
+public boolean polygonContainsPoint(double latitude, double longitude) {
+    String h3Index = h3.latLngToCell(latitude, longitude, R);
+    if (cellsCoveringPolygon.contains(h3Index)) {
+        // The polygon possibly contains the point, so use a more expensive point
+        // in polygon algorithm to determine.
+        return polygonContainsPointExpensive(latitude, longitude);
+    }
+    // The polygon definitely does not contain the point, the expensive check
+    // is not needed.
+    return false;
+}
+```
+
+* Analytics
+
+There is no convenient function to index a polygon into the H3 grid for analytics purposes. Polygons that are small enough, or shaped in specific ways, will not be converted into points, requiring workarouns in order to use them.
+
+* Spherical geometry consistency
+
+Most of the H3 library uses spherical geoemtry. For example, cell boundaries are spherical hexagons
+and pentagons. The `polyfill` function is different that it assumes Cartesian geometry. For
+consistency with the rest of the library, the `polyfill` functions should be able to use the same
+cell boundaries.
+
+* Very large polygons
```

Comment:
Do you have suggestions here? E.g. `maxPolyfillSizes` could return an array of sizes, and `polyfillIncremental(batchNum, polygon, out)` could polyfill a batch? (How this would work in the lib is still a bit fuzzy, but I think it's possible)

Reply:
I was considering if it would be possible to do this in an iterator but haven't looked into this more. We would need some allocation done, but perhaps that could be made relative to the number of points in the polygon rather than the number of cells. A simple polygon producing a large number of cells is related to the problem seen in bytesandbrains/h3-pg#56